### PR TITLE
PPS-588 add guppy csrf

### PR DESCRIPTION
**Pull request by mfshao**
Jira Ticket: [PPS-588](https://ctds-planx.atlassian.net/browse/PPS-588)

### Improvements
- Add CSRF protection for Guppy paths

### Deployment changes
- After merging this, envs must use a data-portal version that contains https://github.com/uc-cdis/data-portal/pull/1464. Otherwise the communication between portal and guppy will be interrupted

[PPS-588]: https://ctds-planx.atlassian.net/browse/PPS-588?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by MaribelleHGomez — DISMISSED** [review]
lgtm

File: kube/services/revproxy/gen3.nginx.conf/guppy-service.conf
Comment (paulineribeyre) [inline kube/services/revproxy/gen3.nginx.conf/guppy-service.conf:2]: i don't think we can't merge this 🤔 like you mentioned in the PR description, all the data commons that don't have the latest data-portal and guppy will fail... we need to somehow make this conditional based on the deployed version of portal and guppy, or leave it out for now and merge it in a few months once everyone has deployed the new portal and guppy.

But iirc we need this for a security ticket? if there's a deadline we might have to make everyone upgrade portal and guppy. Not great right before we all go on break 😬 can it be pushed to Jan, or even Feb when the change is included in 2023.02?

File: kube/services/revproxy/gen3.nginx.conf/guppy-service.conf
Comment (paulineribeyre) [inline kube/services/revproxy/gen3.nginx.conf/guppy-service.conf:3]: maybe this could say "failed csrf check. Make sure to use guppy version >=x and data-portal version >=y"

File: kube/services/revproxy/gen3.nginx.conf/guppy-service.conf
Comment (mfshao) [reply kube/services/revproxy/gen3.nginx.conf/guppy-service.conf:2]: Yeah that is true, the process we have formulized is not to merge this right away. The Guppy and Portal changes will be merged, and then the Portal changes will be cherry-picked into the 2023.12 release (no need to update Guppy since the Guppy changes are purely frontend). And Elise will continue her work on updating envs to ES7, which will bring all these envs to Portal 2023.12. After that we then we will merge this cloud-auto PR and re-roll Portal in those envs

File: kube/services/revproxy/gen3.nginx.conf/guppy-service.conf
Comment (mfshao) [reply kube/services/revproxy/gen3.nginx.conf/guppy-service.conf:3]: Good call, I can update this

**Review by jawadqur — APPROVED** [review]

**Pull request merged**
